server/src/main/java/org/apache/iotdb/db/metrics/dropwizard/reporter/IoTDBReporter.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.metrics.dropwizard.reporter;
+
+import org.apache.iotdb.db.conf.IoTDBConfig;
+import org.apache.iotdb.db.exception.StorageEngineException;
+import org.apache.iotdb.db.exception.metadata.IllegalPathException;
+import org.apache.iotdb.db.exception.metadata.StorageGroupNotSetException;
+import org.apache.iotdb.db.exception.query.QueryProcessException;
+import org.apache.iotdb.db.metadata.path.PartialPath;
+import org.apache.iotdb.db.metrics.metricsUtils;
+import org.apache.iotdb.db.qp.physical.crud.InsertRowPlan;
+import org.apache.iotdb.db.service.basic.BasicServiceProvider;
+import org.apache.iotdb.db.utils.DataTypeUtils;
+import org.apache.iotdb.rpc.IoTDBConnectionException;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+
+import com.codahale.metrics.*;
+import com.codahale.metrics.Timer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class IoTDBReporter extends ScheduledReporter {
+
+  private static final TimeUnit DURATION_UNIT = TimeUnit.MILLISECONDS;
+  private static final TimeUnit RATE_UNIT = TimeUnit.SECONDS;
+  private final String prefix;
+  private static final Logger LOGGER = LoggerFactory.getLogger(IoTDBReporter.class);
+  private BasicServiceProvider basicServiceProvider;
+  private final int rpcPort;
+  private final String address;
+
+  protected IoTDBReporter(
+      MetricRegistry registry,
+      String prefix,
+      MetricFilter filter,
+      ScheduledExecutorService executor,
+      boolean shutdownExecutorOnStop) {
+    super(
+        registry,
+        "iotdb-reporter",
+        filter,
+        RATE_UNIT,
+        DURATION_UNIT,
+        executor,
+        shutdownExecutorOnStop);
+    this.prefix = prefix;
+    IoTDBConfig ioTDBConfig = new IoTDBConfig();
```

Comment:
why not IoTDBDescriptor.getInstance().getConfig()
I thin if we use new IoTDBConfig(), the rpcPort and adress will be 0.0.0.0 and 6667 other than what user configured

Reply:
I've replaced it with IoTDBDescriptor.getInstance().getConfig().